Modules/IO/JPEG/test/itkJPEGImageIODegenerateCasesTest.cxx

```diff
@@ -0,0 +1,52 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkJPEGImageIO.h"
+#include "itkImageFileReader.h"
+#include "itkTestingMacros.h"
+
+
+int
+itkJPEGImageIODegenerateCasesTest(int argc, char * argv[])
+{
+  if (argc != 2)
+  {
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv);
+    std::cerr << " inputFilename" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 2;
+  using PixelType = unsigned char;
+
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  itk::JPEGImageIO::Pointer io = itk::JPEGImageIO::New();
+
+  itk::ImageFileReader<ImageType>::Pointer reader = itk::ImageFileReader<ImageType>::New();
+
+  reader->SetFileName(argv[1]);
+  reader->SetImageIO(io);
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(reader->Update());
```

Comment:
@jhlegarreta  This particular file doesn't trigger exception, so the test works, other broken files can trigger exception, before entering the scan-lines _while_ loop. Important is to avoid segmentation fault or crash, exception is OK. Just FYI.

Reply:
Hats off to the effort to investigate this @issakomi. I cannot investigate as closely. I can only point to a few things:
- I am not sure about the debug/release compilation of the CI builds, so it might well be that the image provided by @Hconk does not trigger any issue in our CI builds. However, many CDash builds are build in release mode, so any (appropriately) corrupted file without this fix should trigger the issue as I understand it.
- If prior to this patch set a given test file was generating a segmentation fault or crash of any nature, then that test file should be added to the test images, and ensure that it does not produce a segfault with this patch set, and is dealt with nicely, either being read if it can be read or generating an exception (and I'd say exception and not warning) saying that it cannot be read. That exception can be tested with the `ITK_TRY_EXPECT_EXCEPTION` macro.
- If the image provided by @Hconk is now being dealt with nicely and can be read without issues, the `ITK_TRY_EXPECT_NO_EXCEPTION` is OK.
- Otherwise, I do not see that this patch set adds new exceptions. If any of the existing ones were not being tested, and we would like them to be tested, images triggering them should be added. See here the exceptions that are not being exercised (any line in red is not exercised/tested):
https://open.cdash.org/viewCoverageFile.php?buildid=7620043&fileid=49347999

Hope this casts some light on what is desired to fully test these cases, and to ensure that the behavior is predictable and conforming to the expectations.

Reply:
The issue with undefined behavior is fixed, incorrect _setjmp_ implementation caused it. It works in debug, release etc. the same way.

> See here the exceptions that are not being exercised (any line in red is not exercised/tested):
> https://open.cdash.org/viewCoverageFile.php?buildid=7620043&fileid=49347999

I can give you an image that will cover this part of code. This one, for example.
https://drive.google.com/file/d/1zGobe2UkAo1nEYMfEv5IV1TM5bhMYx7G/view?usp=sharing

Reply:
OK. Thanks. Then testing the exceptions would be best addressed in a separate PR. I will keep a note and propose a separate PR that uses the shared image (thanks for doing it) as time permits.